src/utils/hooks/useUserSettings/useUserSettings.ts

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from 'react';
+
+import {
+  ConfigMapModel,
+  UserModel,
+  modelToGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1ConfigMap } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  k8sCreate,
+  k8sGet,
+  k8sPatch,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+type UseUserSettings = (
+  key?: string,
+) => [{ [key: string]: any }, (val: any) => void, boolean, Error];
+
+const useUserSettings: UseUserSettings = (key) => {
+  const [error, setError] = useState<Error>();
+  const [userName, setUserName] = useState<string>();
+  const [userSettings, setUserSettings] = useState<{ [key: string]: any }>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchUserName = async () => {
+      setError(null);
+      setLoading(true);
+      try {
+        const user = await k8sGet({ model: UserModel, path: '~' });
+        setUserName(user?.metadata?.name?.replace(':', '-'));
+      } catch (e) {
+        setError(e);
+      }
+      setLoading(false);
+    };
+    fetchUserName();
+  }, []);
+
+  const [userConfigMap, loadedConfigMap, configMapError] =
+    useK8sWatchResource<IoK8sApiCoreV1ConfigMap>(
+      userName && {
+        groupVersionKind: modelToGroupVersionKind(ConfigMapModel),
+        namespace: 'default',
+        name: `${userName}-kubevirt-settings`,
```

Comment:
can you please create a local const for this as it's being used at line 59 as well?